<?xml version="1.0" encoding="UTF-8"?>
<launch>

  <arg name="world" default="$(find hunter_killer_mast)/worlds/droneship_world.world"/> 
  <arg name="paused" default="true"/>
  <arg name="use_sim_time" default="true"/>
  <arg name="gui" default="true"/>
  <arg name="headless" default="false"/>
  <arg name="debug" default="false"/>


  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="paused" value="$(arg paused)"/>
    <arg name="use_sim_time" value="$(arg use_sim_time)"/>
    <arg name="gui" value="$(arg gui)"/>
    <arg name="headless" value="$(arg headless)"/>
    <arg name="debug" value="$(arg debug)"/>
    <!--arg name="world_name" value="$(arg world)"/-->
  </include>
  
    <group ns="/daughter6">
        
        <!-- Robot model -->
        <param name="robot_description" command="$(find xacro)/xacro --inorder '$(find iris)/urdf/daughter6.urdf'" />
        <param name="robot_description_sdf" textfile="$(find iris)/urdf/iris.sdf" />
        <!-- Spawn the robot model -->
        <node name="mybot_spawn_drone" pkg="gazebo_ros" type="spawn_model" output="screen"
              args="-sdf -param robot_description_sdf -x 0 -y -2.5 -z 0.051 -model iris" />
      
        <!-- Load controllers -->
        <rosparam command="load" file="$(find iris)/config/joints.yaml" />
        
        <!-- Controllers -->
        <node name="daughter6_controller_spawner" pkg="controller_manager" type="spawner"
            respawn="false" output="screen" ns="/daughter6"
            args="--namespace=/daughter6
            joint_state_controller
            x_position_controller
            y_position_controller
            z_position_controller
            slider_position_controller
            mast_gripper_position_controller
            left_gripper_position_controller
            right_gripper_position_controller
            telescope_position_controller
            servo_position_controller
            lantenna_position_controller
            rantenna_position_controller
            --timeout 300">
        </node>
        <node name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure" />
        <node name="rqt_publisher" pkg="rqt_publisher" type="rqt_publisher" />

    </group>

    <!--group ns="/hunter_killer_mast">
        
        <param name="robot_description" command="$(find xacro)/xacro inorder '$(find hunter_killer_mast)/urdf/hunter_killer_mast.urdf'" />
        <param name="robot_description_sdf" textfile="$(find hunter_killer_mast)/urdf/hunter_killer_mast.sdf" />
        
        <node name="mybot_spawn_hunter_killer_mast" pkg="gazebo_ros" type="spawn_model" output="screen"
              args="-sdf -param robot_description_sdf -x 0 -y 0.5 -z 0.051 -model hunter_killer_mast" />
      
        
        <rosparam command="load" file="$(find hunter_killer_mast)/config/joints.yaml" />
        
        <node name="hunter_killer_mast_controller_spawner" pkg="controller_manager" type="spawner"
            respawn="false" output="screen" ns="/hunter_killer_mast"
            args="namespace=/hunter_killer_mast
            joint_state_controller_mast
            roll_controller
            pitch_controller
            timeout 300">
        </node>
        

    
    </group-->
          
</launch>